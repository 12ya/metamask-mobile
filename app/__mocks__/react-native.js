import { Linking, DeviceEventEmitter } from 'react-native';

Linking.addEventListener = jest.fn();
Linking.removeEventListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeEventListener
  if (typeof callback === 'function') {
    callback();
  }
});
Linking.openURL = jest.fn();
Linking.canOpenURL = jest.fn();
Linking.getInitialURL = jest.fn();

DeviceEventEmitter.addListener = jest.fn();
DeviceEventEmitter.removeListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeListener
  if (typeof callback === 'function') {
    callback();
  }
});

export default { Linking, DeviceEventEmitter };
