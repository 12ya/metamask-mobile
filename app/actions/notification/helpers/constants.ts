enum notificationsAction {
  PERFORM_SIGN_IN_REQUEST = 'notifications/perform_sign_in_request',
  PERFORM_SIGN_IN_SUCCESS = 'notifications/perform_sign_in_success',
  PERFORM_SIGN_IN_FAILURE = 'notifications/perform_sign_in_failure',
  PERFORM_SIGN_OUT_REQUEST = 'notifications/perform_sign_out_request',
  PERFORM_SIGN_OUT_SUCCESS = 'notifications/perform_sign_out_success',
  PERFORM_SIGN_OUT_FAILURE = 'notifications/perform_sign_out_failure',
  ENABLE_PROFILE_SYNCING_REQUEST = 'notifications/enable_profile_syncing_request',
  ENABLE_PROFILE_SYNCING_SUCCESS = 'notifications/enable_profile_syncing_success',
  ENABLE_PROFILE_SYNCING_FAILURE = 'notifications/enable_profile_syncing_failure',
  DISABLE_PROFILE_SYNCING_REQUEST = 'notifications/disable_profile_syncing_request',
  DISABLE_PROFILE_SYNCING_SUCCESS = 'notifications/disable_profile_syncing_success',
  DISABLE_PROFILE_SYNCING_FAILURE = 'notifications/disable_profile_syncing_failure',
  ENABLE_PUSH_NOTIFICATIONS_REQUEST = 'notifications/enable_push_notifications_request',
  ENABLE_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/enable_push_notifications_success',
  ENABLE_PUSH_NOTIFICATIONS_FAILURE = 'notifications/enable_push_notifications_failure',
  DISABLE_PUSH_NOTIFICATIONS_REQUEST = 'notifications/disable_push_notifications_request',
  DISABLE_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/disable_push_notifications_success',
  DISABLE_PUSH_NOTIFICATIONS_FAILURE = 'notifications/disable_push_notifications_failure',
  CHECK_ACCOUNTS_PRESENCE_REQUEST = 'notifications/check_accounts_presence_request',
  CHECK_ACCOUNTS_PRESENCE_SUCCESS = 'notifications/check_accounts_presence_success',
  CHECK_ACCOUNTS_PRESENCE_FAILURE = 'notifications/check_accounts_presence_failure',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST = 'notifications/delete_on_chain_triggers_by_account_request',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS = 'notifications/delete_on_chain_triggers_by_account_success',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE = 'notifications/delete_on_chain_triggers_by_account_failure',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST = 'notifications/update_on_chain_triggers_by_account_request',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS = 'notifications/update_on_chain_triggers_by_account_success',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE = 'notifications/update_on_chain_triggers_by_account_failure',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_REQUEST = 'notifications/set_feature_announcements_enabled_request',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_SUCCESS = 'notifications/set_feature_announcements_enabled_success',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_FAILURE = 'notifications/set_feature_announcements_enabled_failure',
  SET_SNAP_NOTIFICATIONS_ENABLED_REQUEST = 'notifications/set_snap_notifications_enabled_request',
  SET_SNAP_NOTIFICATIONS_ENABLED_SUCCESS = 'notifications/set_snap_notifications_enabled_success',
  SET_SNAP_NOTIFICATIONS_ENABLED_FAILURE = 'notifications/set_snap_notifications_enabled_failure',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_REQUEST = 'notifications/set_metamask_notifications_feature_seen_request',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_SUCCESS = 'notifications/set_metamask_notifications_feature_seen_success',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_FAILURE = 'notifications/set_metamask_notifications_feature_seen_failure',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_REQUEST = 'notifications/fetch_and_update_metamask_notifications_request',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_SUCCESS = 'notifications/fetch_and_update_metamask_notifications_success',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_FAILURE = 'notifications/fetch_and_update_metamask_notifications_failure',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_REQUEST = 'notifications/mark_metamask_notifications_as_read_request',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_SUCCESS = 'notifications/mark_metamask_notifications_as_read_success',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_FAILURE = 'notifications/mark_metamask_notifications_as_read_failure',
  DELETE_NOTIFICATION_STATUS_REQUEST = 'notifications/delete_notification_status_request',
  DELETE_NOTIFICATION_STATUS_SUCCESS = 'notifications/delete_notification_status_success',
  DELETE_NOTIFICATION_STATUS_FAILURE = 'notifications/delete_notification_status_failure',
  SET_PARTICIPATE_IN_META_METRICS_REQUEST = 'notifications/set_participate_in_meta_metrics_request',
  SET_PARTICIPATE_IN_META_METRICS_SUCCESS = 'notifications/set_participate_in_meta_metrics_success',
  SET_PARTICIPATE_IN_META_METRICS_FAILURE = 'notifications/set_participate_in_meta_metrics_failure',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_REQUEST = 'notifications/update_trigger_push_notifications_request',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/update_trigger_push_notifications_success',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_FAILURE = 'notifications/update_trigger_push_notifications_failure',
  ENABLE_NOTIFICATIONS_SERVICES_REQUEST = 'notifications/enable_notifications_services_request',
  ENABLE_NOTIFICATIONS_SERVICES_SUCCESS = 'notifications/enable_notifications_services_success',
  ENABLE_NOTIFICATIONS_SERVICES_FAILURE = 'notifications/enable_notifications_services_failure',
  DISABLE_NOTIFICATIONS_SERVICES_REQUEST = 'notifications/disable_notifications_services_request',
  DISABLE_NOTIFICATIONS_SERVICES_SUCCESS = 'notifications/disable_notifications_services_success',
  DISABLE_NOTIFICATIONS_SERVICES_FAILURE = 'notifications/disable_notifications_services_failure',
}

export default notificationsAction;
