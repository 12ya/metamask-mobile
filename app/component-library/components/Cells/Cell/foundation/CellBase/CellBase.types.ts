// External dependencies.
import { AvatarProps } from '../../../../Avatars/Avatar/Avatar.types';
import { ListItemProps } from '../../../../List/ListItem/ListItem.types';

/**
 * Cell Account component props.
 */
export interface CellBaseProps extends ListItemProps {
  /**
   * Props for avatar component (with the exception of size).
   * Avatar size is defaulted to size Md (32x32) for Cells
   */
  avatarProps: AvatarProps;
  /**
   * Title of the Cell Account, 1 line truncation.
   */
  title: string | React.ReactNode;
  /**
   * Optional secondary text below the title, 1 line truncation.
   */
  secondaryText?: string | React.ReactNode;
  /**
   * Optional tertiary text below the secondaryText, 1 line truncation.
   */
  tertiaryText?: string | React.ReactNode;
  /**
   * Optional label (using Tag component) below the title/secondaryText/tertiaryText.
   */
  tagLabel?: string;
  /**
   * Optional accessory that can be inserted on the right of Cell Account.
   */
  children?: React.ReactNode;
}

/**
 * Style sheet input parameters.
 */
export type CellBaseStyleSheetVars = Pick<CellBaseProps, 'style'>;
