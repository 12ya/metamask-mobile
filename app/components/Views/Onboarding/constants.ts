// eslint-disable-next-line import/prefer-default-export
export const onboardingStrategy = {
  privateKey: 'privateKey',
  seedPhrase: 'seedPhrase',
  qrHardware: 'qrHardware',
};
