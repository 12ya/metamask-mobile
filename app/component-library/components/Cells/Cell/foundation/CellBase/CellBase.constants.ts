// External dependencies
import { SAMPLE_AVATAR_PROPS } from '../../../../Avatars/Avatar/Avatar.constants';
import { AvatarSize } from '../../../../Avatars/Avatar';
import { TextVariant, TextColor } from '../../../../Texts/Text';
import { VerticalAlignment } from '../../../../List/ListItem';

// Internal dependencies
import { CellBaseProps } from './CellBase.types';

// Defaults
export const DEFAULT_CELLBASE_AVATAR_SIZE = AvatarSize.Md;
export const DEFAULT_CELLBASE_TITLE_TEXTVARIANT = TextVariant.HeadingSMRegular;
export const DEFAULT_CELLBASE_TITLE_TEXTCOLOR = TextColor.Default;
export const DEFAULT_CELLBASE_SECONDARYTEXT_TEXTVARIANT = TextVariant.BodyMD;
export const DEFAULT_CELLBASE_SECONDARYTEXT_TEXTCOLOR = TextColor.Alternative;
export const DEFAULT_CELLBASE_TERTIARYTEXT_TEXTVARIANT = TextVariant.BodyMD;
export const DEFAULT_CELLBASE_TERTIARYTEXT_TEXTCOLOR = TextColor.Alternative;
export const DEFAULT_CELLBASE_LISTITEM_GAP = 16;
export const DEFAULT_CELLBASE_LISTITEM_VERTICALALIGNMENT =
  VerticalAlignment.Center;

// Sample consts
const SAMPLE_CELLBASE_TITLE = 'Orangefox.eth';
const SAMPLE_CELLBASE_SECONDARYTEXT =
  '0x2990079bcdEe240329a520d2444386FC119da21a';
const SAMPLE_CELLBASE_TERTIARY_TEXT = 'Updated 1 sec ago';
const SAMPLE_CELLBASE_TAGLABEL = 'Imported';

export const SAMPLE_CELLBASE_PROPS: CellBaseProps = {
  avatarProps: SAMPLE_AVATAR_PROPS,
  title: SAMPLE_CELLBASE_TITLE,
  secondaryText: SAMPLE_CELLBASE_SECONDARYTEXT,
  tertiaryText: SAMPLE_CELLBASE_TERTIARY_TEXT,
  tagLabel: SAMPLE_CELLBASE_TAGLABEL,
};
