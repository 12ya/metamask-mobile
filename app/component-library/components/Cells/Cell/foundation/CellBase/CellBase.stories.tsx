// Third party dependencies
import React from 'react';
import { Meta, Story } from '@storybook/react-native';

// Internal dependencies
import CellBase from './CellBase';
import { CellBaseProps } from './CellBase.types';
import { SAMPLE_CELLBASE_PROPS } from './CellBase.constants';

export default {
  title: 'Component Library / Cells / CellBase',
  component: CellBase,
  argTypes: {
    title: { control: 'text' },
    secondaryText: { control: 'text' },
    tertiaryText: { control: 'text' },
    tagLabel: { control: 'text' },
  },
} as Meta;

const Template: Story<CellBaseProps> = (args) => <CellBase {...args} />;

export const Default = Template.bind({});
Default.args = {
  ...SAMPLE_CELLBASE_PROPS,
};
