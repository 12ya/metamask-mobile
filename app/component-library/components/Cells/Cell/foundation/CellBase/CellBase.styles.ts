// Third library dependencies.
import { StyleSheet, ViewStyle } from 'react-native';

// External dependencies.
import { CellBaseStyleSheetVars } from './CellBase.types';

// Internal dependencies.
import { Theme } from '../../../../../../util/theme/models';

/**
 * Style sheet function for CellBase component.
 *
 * @param params Style sheet params.
 * @param params.theme App theme from ThemeContext.
 * @param params.vars Inputs that the style sheet depends on.
 * @returns StyleSheet object.
 */
const styleSheet = (params: { theme: Theme; vars: CellBaseStyleSheetVars }) => {
  const { vars } = params;
  const { style } = vars;

  return StyleSheet.create({
    cellBase: Object.assign({ padding: 0 } as ViewStyle, style) as ViewStyle,
    tagLabel: {
      alignSelf: 'flex-start',
      marginTop: 4,
    },
  });
};

export default styleSheet;
