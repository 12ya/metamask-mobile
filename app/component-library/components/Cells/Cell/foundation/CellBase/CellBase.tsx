/* eslint-disable react/prop-types */

// Third library dependencies.
import React from 'react';

// External dependencies
import { useStyles } from '../../../../../hooks';
import Tag from '../../../../Tags/Tag';
import Avatar from '../../../../Avatars/Avatar';

// External dependencies
import ListItem from '../../../../List/ListItem';
import ListItemColumn, { WidthType } from '../../../../List/ListItemColumn';
import renderStringOrNode from '../../../../../utility/renderStringOrNode';

// Internal dependencies
import {
  DEFAULT_CELLBASE_AVATAR_SIZE,
  DEFAULT_CELLBASE_TITLE_TEXTVARIANT,
  DEFAULT_CELLBASE_TITLE_TEXTCOLOR,
  DEFAULT_CELLBASE_SECONDARYTEXT_TEXTVARIANT,
  DEFAULT_CELLBASE_SECONDARYTEXT_TEXTCOLOR,
  DEFAULT_CELLBASE_TERTIARYTEXT_TEXTVARIANT,
  DEFAULT_CELLBASE_TERTIARYTEXT_TEXTCOLOR,
  DEFAULT_CELLBASE_LISTITEM_GAP,
  DEFAULT_CELLBASE_LISTITEM_VERTICALALIGNMENT,
} from './CellBase.constants';
import styleSheet from './CellBase.styles';
import { CellBaseProps } from './CellBase.types';
import { CellModalSelectorsIDs } from '../../../../../../../e2e/selectors/Modals/CellModal.selectors';

const CellBase = ({
  style,
  avatarProps,
  title,
  secondaryText,
  tertiaryText,
  tagLabel,
  children,
}: CellBaseProps) => {
  const { styles } = useStyles(styleSheet, { style });

  return (
    <ListItem
      style={styles.cellBase}
      gap={DEFAULT_CELLBASE_LISTITEM_GAP}
      verticalAlignment={DEFAULT_CELLBASE_LISTITEM_VERTICALALIGNMENT}
    >
      {/* DEV Note: Account Avatar should be replaced with Avatar with Badge whenever available */}
      <ListItemColumn>
        <Avatar
          testID={CellModalSelectorsIDs.BASE_AVATAR}
          size={DEFAULT_CELLBASE_AVATAR_SIZE}
          {...avatarProps}
        />
      </ListItemColumn>
      <ListItemColumn widthType={WidthType.Fill}>
        {renderStringOrNode(title, {
          numberOfLines: 1,
          variant: DEFAULT_CELLBASE_TITLE_TEXTVARIANT,
          color: DEFAULT_CELLBASE_TITLE_TEXTCOLOR,
          testID: CellModalSelectorsIDs.BASE_TITLE,
        })}
        {secondaryText &&
          renderStringOrNode(secondaryText, {
            numberOfLines: 1,
            variant: DEFAULT_CELLBASE_SECONDARYTEXT_TEXTVARIANT,
            color: DEFAULT_CELLBASE_SECONDARYTEXT_TEXTCOLOR,
          })}
        {tertiaryText &&
          renderStringOrNode(tertiaryText, {
            numberOfLines: 1,
            variant: DEFAULT_CELLBASE_TERTIARYTEXT_TEXTVARIANT,
            color: DEFAULT_CELLBASE_TERTIARYTEXT_TEXTCOLOR,
          })}
        {!!tagLabel && <Tag label={tagLabel} style={styles.tagLabel} />}
      </ListItemColumn>
      {children && <ListItemColumn>{children}</ListItemColumn>}
    </ListItem>
  );
};

export default CellBase;
