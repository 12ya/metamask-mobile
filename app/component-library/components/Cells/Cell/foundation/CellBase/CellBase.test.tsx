// Third party dependencies.
import React from 'react';
import { render } from '@testing-library/react-native';
import { View } from 'react-native';

// Internal dependencies.
import CellBase from './CellBase';
import { SAMPLE_CELLBASE_PROPS } from './CellBase.constants';
import { CellModalSelectorsIDs } from '../../../../../../../e2e/selectors/Modals/CellModal.selectors';

describe('CellBase', () => {
  describe('CellBase Component', () => {
    const renderComponent = (props = {}) =>
      render(<CellBase {...SAMPLE_CELLBASE_PROPS} {...props} />);
    const SAMPLEVIEW_TESTID = 'sampleview';
    const SampleView = <View testID={SAMPLEVIEW_TESTID} />;

    it('should render CellBase component', () => {
      const { toJSON, queryByTestId } = renderComponent();
      expect(toJSON()).toMatchSnapshot();
      expect(queryByTestId(CellModalSelectorsIDs.BASE_AVATAR)).not.toBe(null);
      expect(queryByTestId(CellModalSelectorsIDs.BASE_TITLE)).not.toBe(null);
    });

    it('should render with correct title', () => {
      const { getByText } = renderComponent();
      expect(getByText(SAMPLE_CELLBASE_PROPS.title as string)).toBeDefined();
    });

    it('should render with correct title node', () => {
      const { getByTestId } = renderComponent({ title: SampleView });
      expect(getByTestId(SAMPLEVIEW_TESTID)).toBeDefined();
    });

    it('should render with correct secondaryText', () => {
      const { getByText } = renderComponent();
      expect(
        getByText(SAMPLE_CELLBASE_PROPS.secondaryText as string),
      ).toBeDefined();
    });

    it('should render with correct secondaryText node', () => {
      const { getByTestId } = renderComponent({ secondaryText: SampleView });
      expect(getByTestId(SAMPLEVIEW_TESTID)).toBeDefined();
    });

    it('should render with correct tertiaryText', () => {
      const { getByText } = renderComponent();
      expect(
        getByText(SAMPLE_CELLBASE_PROPS.tertiaryText as string),
      ).toBeDefined();
    });

    it('should render with correct tertiaryText node', () => {
      const { getByTestId } = renderComponent({ tertiaryText: SampleView });
      expect(getByTestId(SAMPLEVIEW_TESTID)).toBeDefined();
    });

    it('should render with correct tagLabel', () => {
      const { getByText } = renderComponent();
      expect(getByText(SAMPLE_CELLBASE_PROPS.tagLabel as string)).toBeDefined();
    });

    it('should render children correctly', () => {
      const { getByTestId } = renderComponent({ children: SampleView });
      expect(getByTestId(SAMPLEVIEW_TESTID)).toBeDefined();
    });
  });
});
