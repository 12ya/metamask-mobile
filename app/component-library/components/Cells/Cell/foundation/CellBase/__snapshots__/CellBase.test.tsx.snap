// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellBase - Snapshot should render default settings correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Avatar
    jazzIconProps={
      Object {
        "address": "0x10e08af911f2e489480fb2855b24771745d0198b50f5c55891369844a8c57092",
      }
    }
    size="32"
    style={
      Object {
        "marginRight": 16,
      }
    }
    testID="cell-avatar"
    variant="JazzIcon"
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <Text
      numberOfLines={1}
      testID="cell-title"
      variant="sHeadingSMRegular"
    >
      Orangefox.eth
    </Text>
  </View>
</View>
`;

exports[`CellBase - Snapshot should render label when given 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Avatar
    jazzIconProps={
      Object {
        "address": "0x10e08af911f2e489480fb2855b24771745d0198b50f5c55891369844a8c57092",
      }
    }
    size="32"
    style={
      Object {
        "marginRight": 16,
      }
    }
    testID="cell-avatar"
    variant="JazzIcon"
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <Text
      numberOfLines={1}
      testID="cell-title"
      variant="sHeadingSMRegular"
    >
      Orangefox.eth
    </Text>
    <Tag
      label="Imported"
      style={
        Object {
          "marginTop": 4,
        }
      }
      testID="cell-label"
    />
  </View>
</View>
`;

exports[`CellBase - Snapshot should render secondaryText when given 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Avatar
    jazzIconProps={
      Object {
        "address": "0x10e08af911f2e489480fb2855b24771745d0198b50f5c55891369844a8c57092",
      }
    }
    size="32"
    style={
      Object {
        "marginRight": 16,
      }
    }
    testID="cell-avatar"
    variant="JazzIcon"
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <Text
      numberOfLines={1}
      testID="cell-title"
      variant="sHeadingSMRegular"
    >
      Orangefox.eth
    </Text>
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#535A61",
        }
      }
      testID="cell-secondary-text"
      variant="sBodyMD"
    >
      0x2990079bcdEe240329a520d2444386FC119da21a
    </Text>
  </View>
</View>
`;

exports[`CellBase - Snapshot should render tertiaryText when given 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Avatar
    jazzIconProps={
      Object {
        "address": "0x10e08af911f2e489480fb2855b24771745d0198b50f5c55891369844a8c57092",
      }
    }
    size="32"
    style={
      Object {
        "marginRight": 16,
      }
    }
    testID="cell-avatar"
    variant="JazzIcon"
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <Text
      numberOfLines={1}
      testID="cell-title"
      variant="sHeadingSMRegular"
    >
      Orangefox.eth
    </Text>
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#535A61",
        }
      }
      testID="cell-tertiary-text"
      variant="sBodyMD"
    >
      Updated 1 sec ago
    </Text>
  </View>
</View>
`;
